canonical-web-and-design/charmhub.io
# Add GA click events to navigation.js

## Done

* GA event framework to capture:
  * Nav and footer usage
  * Link clicks in content (charm cards, listing navigation, etc.)

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://localhost:8045
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Open GA on the charmhub.io or combo tracker property -> Real time view -> Refine to yourself using location filtering if possible -> Events. Otherwise, you can use the GA debugger [(Chrome/Chromium only)](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna) and look at console output.
- Ensure the charmhub.io cookie policy is accepted at least at the "performance" level.
- Use the site: ensure clicking on charm card is sending a card event, click links, use the main navigation, navigate between tabs of a listing.

## Screenshots

![Screenshot from 2020-11-18 13-02-32](https://user-images.githubusercontent.com/18480003/99528318-66ba7780-299e-11eb-9dfc-4c1d47093bcf.png)

![Screenshot from 2020-11-18 13-03-24](https://user-images.githubusercontent.com/18480003/99528357-776aed80-299e-11eb-963f-e3e19da7e297.png)
